include/endpoint/Endpoint.h

```diff
@@ -1,16 +1,19 @@
 #pragma once
 #include <pistache/http.h>
 #include <pistache/endpoint.h>
+#include <util/config.h>
 
 namespace endpoint {
     class Endpoint {
     public:
+        Endpoint() = default;
+
         Endpoint(int port, int threadCount);
         void start();
 
         virtual std::shared_ptr<Pistache::Http::Handler> getHandler() = 0;
     private:
-        int port;
-        int threadCount;
+        int port = util::getSystemConfig().endpointPort;
+        int threadCount = util::getSystemConfig().endpointNumThreads;
```

Comment:
Can these be set in an initialiser list on the constructor implementation rather than in this header?

Reply:
No because there is more than one constructor: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-in-class-initializer

You also always want a constructor in which you can pass them in for testing, same for the subclasses. 
There is a very good CPPconf (2018?) video about how this is this too that agrees with Bjarne Stroustrup on this.

Reply:
Is that link not talking about constants though? I'm also not sure that having more than one constructor means you shouldn't use initialiser lists either. These are variables that will be changed by different system configurations and different constructors.

For checking changes in the system config in tests, we modify the SystemConfig singleton or corresponding environment variable (arguably replicating the real behaviour of the system more, but breaking from a strict unit test). Therefore I'm not sure I buy the testing angle.

I'm not that bothered by it, it just seems out of sync with the constructor and testing style in the rest of the codebase (which of course may be wrong). I'm inclined to stick with consistency and make wholesale changes if we want to change how we do something rather than have things out of sync.